doc/sonic-build-system/SONiC-Reproduceable-Build.md

```diff
@@ -0,0 +1,479 @@
+
+The tarball can be generated by the following command:
+
+debootstrap --variant=minbase --arch amd64 --make-tarball=tarball.tgz buster rootfs
+
+We can generate the tarball and replace the all packages with the right versions, or generate the tarball ourselves to support the reproduceable build.
+
+The tarball format is as below:
+
+| File                    | Description                                          |
+| :---------------------- | :--------------------------------------------------- |
+| /debootstrap/base       | The base debian packages                             |
+| /debootstrap/required   | The required debian packages                         |
+| /debootstrap/debpaths   | The debian package name to package real path mapping |
+| /var/cache/apt/archives | Contains the debian packages in the folder           |
+
+The steps to do the reproduceable build for root filesystem.
+
+1. Build the normal root filesystem
+
+   debootstrap --variant=minbase --arch amd64 buster rootfs
+
+2. Freeze the versions, the version configuration format, see [here](#Pypi-Version-Configuration)
+
+3. Download the debian packages based on the frozen versions, and make the debootstrap tarball
+
+4. build the root filesystem using the tarball with frozen debian package versions.
+
+Another approach for debian base image reproduceable build is by [APT-TOOL](https://github.com/pauldotknopf/apt-tool). It supports to freeze the versions, and use the versions in the subsequent builds.
+
+
+# Packages downloaded by wget/curl
+
+The web site may be not stable, and the web packages may be replaced to another one unexpectedly, so we do not want to depend on the web packages at a remote web site. Although there is no version for a package downloaded by wget/curl, the hash value of the package file can be used as the version value. Like Pypi packages and the Debian packages, there is a configuration file for wget/curl. For the package with the same URL, it is expected the same binary will be retrieved when using wget/curl.
+
+Requirements:
+
+1. Use the same web package, even the remote web package changed.
+2. Improve reliability, the build does not be impacted when the remote web site is not available.
+3. Keep update, automatically send a version change Pull Request to use the latest web package. [Low Priority]
+
+## Web Package Version Configuration
+
+The version file in files/build/versions/ for web packages.
+
+| File Name | Description |
+| --- | --- |
+| versions-web | The versions for packages downloaded from web |
+
+Like pip or debian package version configuration, the version format for web package is as below:
+
+<Package URL> **==** <hash value>;
+
+An example as below:
+
+https://storage.googleapis.com/golang/go1.14.2.linux-amd64.tar.gz== ebef065e4d35572af5b03e2be957a9c6c5063b38

```

Comment:
Seems that this is a sha1 hash,
```
$ wget https://storage.googleapis.com/golang/go1.14.2.linux-amd64.tar.gz
$ sha1sum go1.14.2.linux-amd64.tar.gz
ebef065e4d35572af5b03e2be957a9c6c5063b38  go1.14.2.linux-amd64.tar.gz
$ md5sum go1.14.2.linux-amd64.tar.gz
856d248e3ea8a287d13e5f6afd086282  go1.14.2.linux-amd64.tar.gz
```
but [get_url_version](https://github.com/Azure/sonic-buildimage/blob/3db6572e6785ffcd4ef0a4575b16f95c7606c327/src/sonic-build-hooks/scripts/buildinfo_base.sh#L49) uses the md5 hash.

Could you please add explicit note, that this is the md5 hash?